tests/functional/object/mcg/test_bucket_policy.py

```diff
@@ -850,6 +874,202 @@ def test_bucket_policy_verify_invalid_scenarios(
+                    bucketname=obc_obj.bucket_name,
+                    object_key="index.html",
+                    data=index,
+                    content_type="text/html",
+                )
+                raise UnexpectedBehaviour(
+                    "Failed: Completed put object to bucket, expect to get AccessDenied."
+                )
+            except boto3exception.ClientError as e:
+                logger.info(e.response)
+                response = HttpResponseParser(e.response)
+                if response.error["Code"] == "AccessDenied":
+                    logger.info(f"PutObject got {response.error['Code']}")
+                else:
+                    raise UnexpectedBehaviour(
+                        f"{e.response} received invalid error code "
+                        f"{response.error['Code']}"
+                    )
+
+            # Delete bucket is allowed.
+            logger.info(f"Deleting bucket {obc_obj.bucket_name}")
+            assert s3_delete_bucket_website(
+                s3_obj=obc_obj, bucketname=obc_obj.bucket_name
+            ), "Failed to delete bucket."
+
+        # Delete policy and confirm policy got deleted.
+        delete_bucket_policy_verify(obc_obj, mcg_obj)
+
+    @pytest.mark.polarion_id("OCS-5770")
+    @skipif_ocs_version("<4.16")
+    @tier1
+    def test_bucket_policy_elements_NotResource(self, mcg_obj, bucket_factory):
+        """
+        Test bucket policy element of NotResource with Effect: Deny
+        """
+
+        # Creating obc and obc object to get account details, keys etc
+        obc_bucket = bucket_factory(amount=1, interface="OC")
+        obc_obj = OBC(obc_bucket[0].name)
+
+        # Create data and object key
+        data = "Sample string content to write to a new S3 object"
+        object_key = "ObjKey-" + str(uuid.uuid4().hex)
+
+        # Set bucket policy for user
+        bucket_policy_generated = gen_bucket_policy(
+            user_list=obc_obj.obc_account,
+            actions_list=["*"],
+            resources_list=[obc_obj.bucket_name, f'{obc_obj.bucket_name}/{"*"}'],
+            resource="NotResource",
+            effect="Deny",
+        )
+        bucket_policy = json.dumps(bucket_policy_generated)
+
+        # Add Bucket Policy
+        logger.info(f"Creating bucket policy on bucket: {obc_obj.bucket_name}")
+        put_bucket_policy(mcg_obj, obc_obj.bucket_name, bucket_policy)
+
+        # Get bucket policy
+        logger.info(f"Getting Bucket policy on bucket: {obc_obj.bucket_name}")
+        get_policy = get_bucket_policy(mcg_obj, obc_obj.bucket_name)
+        logger.info(f"Got bucket policy: {get_policy['Policy']}")
+
+        # Verify S3 action (putObject) is allowed.
+        logger.info(
+            f"Adding object on the bucket: {obc_obj.bucket_name} using user: {obc_obj.obc_account}"
+        )
+        assert s3_put_object(
+            obc_obj, obc_obj.bucket_name, object_key, data
+        ), "Failed to put Object"
+
```

Comment:
Please also test that actions on other resources are denied as expected. You'll probably have to add two prefixes to the test case: one which you'll exclude from the denial and one which should still be denied.

Reply:
This can be done in the next PR to add more check on this scenario.